tao_compiler/mlir/disc/transforms/disc_transform_schedule.h

```diff
@@ -0,0 +1,130 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef DISC_TRANSFORMS_TRANSFORM_SCHEDULE_H_
+#define DISC_TRANSFORMS_TRANSFORM_SCHEDULE_H_
+
+#include <string>
+#include <unordered_map>
+
+#include "tensorflow/compiler/mlir/disc/transforms/fusion_utils.h"
+
+namespace mlir {
+namespace disc_ral {
+
+// PatternKind reprensets the category of a given schedule.
+// For the same `PatternKind`, we may still have different schedule strategies
+// for different shape range. We further use different tags to distinguish such
+// schedules within the same category.
+enum class PatternKind : int32_t { kNone, kGEMM };
+
+// Converts a pattern kind to its string representation.
+std::string patternKindToString(PatternKind kind);
+
+// Creates a pattern kind from its string representation.
+PatternKind patternKindFromString(const std::string& str);
+
+// PatternDescription collects everything needed to assign schedule for a give
+// fusion pattern.
+class PatternDescription {
+ public:
+  explicit PatternDescription(lmhlo::FusionOp op, FusionPattern& fusionPattern,
+                              ShapeAnalysis& shapeAnalysis);
+
+  // Returns the kind of this `PatternDescription`.
+  PatternKind getPatternKind() const;
+
+  // Returns the tags attached to this fusion pattern.
```

Comment:
what's the meaning of `tag`?

Reply:
see the comment line 27 in the same file.